orca-extensionpoint/src/main/java/com/netflix/spinnaker/orca/extensionpoint/pipeline/ExecutionPreprocessor.java

```diff
@@ -26,10 +26,15 @@
   /**
    * Returns whether or not the preprocess can handle the inbound execution.
    */
-  boolean supports(@Nonnull Map<String, Object> execution);
+  boolean supports(@Nonnull Map<String, Object> execution, @Nonnull Type type);
 
   /**
    * Allows modification of an execution configuration.
    */
   @Nonnull Map<String, Object> process(@Nonnull Map<String, Object> execution);
+
+  enum Type {
+    PIPELINE,
+    ORCHESTRATION
+  }
```

Comment:
I explicitly left a type off this to support cases where type isn't applicable. Maybe add an `ANY`?

Reply:
The PR passes in a `Type` at the call site so I think it's appropriate to be either `PIPELINE` or `ORCHESTRATION` (vs the preprocessor declaring what type it supports).